x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotRecoveryStateIntegrationTests.java

```diff
@@ -216,13 +319,38 @@ public void testFilesStoredInThePersistentCacheAreMarkedAsReusedInRecoveryState(
         }
     }
 
+    private CountDownLatch fillPreWarmThreadPool(String node) throws InterruptedException, BrokenBarrierException {
+        final ThreadPool threadPool = internalCluster().getInstance(ThreadPool.class, node);
+        final int preWarmThreads = threadPool.info(SearchableSnapshotsConstants.CACHE_PREWARMING_THREAD_POOL_NAME).getMax();
+        final Executor executor = threadPool.executor(SearchableSnapshotsConstants.CACHE_PREWARMING_THREAD_POOL_NAME);
+        final CyclicBarrier barrier = new CyclicBarrier(preWarmThreads + 1);
+        final CountDownLatch latch = new CountDownLatch(1);
+        for (int i = 0; i < preWarmThreads; i++) {
+            executor.execute(() -> {
+                try {
+                    barrier.await();
```

Comment:
Is this barrier necessary? Even if the prewarm should put new tasks on the queue, they will go after the tasks you already submitted here.

Reply:
The idea was to ensure that all threads are being executed before we return from this method, so we can ensure that the ThreadPool is full before releasing the `CountDownLatch`